src/utils/version-patch.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+/* eslint-disable no-plusplus */
+import { apply } from './json-patch';
+import { clone } from './data';
+
+/**
+ * Checks if a JSON file (typically index.json or a RenderNode) has version data
+ * @param  {JSON} data to be inspected for version data
+ * @returns {Boolean}
+ */
+function dataHasVersion(data) {
+  if (!data) return false;
+  if (!('metadata' in data)) return false;
+  if (!('version' in data.metadata)) return false;
+  return true;
+}
```

Comment:
You can simplify it with 

```suggestion
function dataHasVersion({ metadata }) {
  return metadata && metadata.version;
}
```